### DISCUSSION TITLE
How exactly does time-series AverageModel work?

### DISCUSSION THREAD

**Discussion by aaronschiff** [autogluon/autogluon#4004] [Q&A] [answered]
The documentation for the time-series AverageModel says it is a "Baseline model that sets the forecast equal to the historic average or quantile.". My understanding is the forecast would be a constant equal to the mean (average) of the past values of the data series. However, that is not what I seem to get from testing simple examples, and I'm confused about how AverageModel works. Here's an example: 

```
import pandas as pd
from autogluon.timeseries import TimeSeriesDataFrame, TimeSeriesPredictor

dates = pd.date_range(start="1990-01-01", periods=8, freq="AS-JAN")

dat = pd.DataFrame(
    {"date": dates, "value": [400, 200, 300, 100, 200, 300, 100, 200]}
).assign(id="test")

print(dat["value"].mean())

dat_ts = TimeSeriesDataFrame.from_data_frame(
    dat, id_column="id", timestamp_column="date"
)

predictor = TimeSeriesPredictor(
    prediction_length=1, path="test", target="value", eval_metric="MASE"
)

predictor.fit(dat_ts, hyperparameters={"AverageModel": {}})

forecasts = predictor.predict(dat_ts)

print(forecasts)
```

This produces a forecast value of 200, but the mean of all the data values is 225. I understand that TimeSeriesPredictor.fit() splits the data into training and test sets, and trains the model on the training set. However, dropping any number of values from the end of the data series in the example above does not give a mean of 200. For example if the last point is dropped, the mean is 228.6. 

Can someone please help me to understand what is happening, or what am I misunderstanding about how these forecasts are created? Thanks!

**Comment by aaronschiff** [comment, marked as answer]
Oh, I figured it out. The forecast is the median of the historic values, not the mean. I was confused by the documentation for AverageModel and the "mean" label in the output from `TimeSeriesPredictor.fit()`.

**Reply by shchur** [reply]
Yes, `"mean"` column always contains the forecast that optimizes the chosen point forecast metric. In this case, since `eval_metric="MASE"` it's the median. If you keep the default value of the metric or set it to something like RMSE, the `"mean"` column would contain the average.

**Reply by aaronschiff** [reply]
Thanks for that detail. I wrongly assumed that no forecast optimisation is done with `AverageModel`.

**Reply by aaronschiff** [reply]
And a follow-up comment to note that when I initially changed `eval_metric="RMSE"`, I saw no change in the forecast `"mean"` output. Eventually I realised this is because the model was cached from the previous fit with `eval_metric="MASE"`, and although I had fit the model again with a different `eval_metric`, the cached model was not updated before the forecasts were generated. Deleting the cache directory and fitting/predicting again produced the expected result with the `"mean"` forecast equal to the mean of the training data. 

In summary it seems to be necessary to either set `cache_predictions=False` when creating a `TimeSeriesPredictor`, or delete the cache when changing the options for local models (if there are no changes to the training data), so that forecasts reflect the updated models. Just noting this here in case it's useful to anyone else.
